docs/versioning.rst

```diff
@@ -0,0 +1,91 @@
+
+.. _versioning:
+
+Versioning
+==========
+
+µsort tries to maintain the *spirit* of `SemVer <https://semver.org/>`_ while
+focusing primarily on the sorting and formatting results than strictly on the
+API contract.
+
+In general, µsort will attempt to maintain consistent formatting results within
+a major version, with a best-effort attempt to remain "backwards compatible"
+with previous versions of µsort within that major release. Except for obvious
+or egregious formatting errors, any file sorted by a newer release of µsort
+should produce no changes if formatted by previous minor/patch versions
+in that family. Our goal is to enable a safe and consistent rollout of new
+sorting or formatting behavior within minor version releases, before updating
+CI or toolchains to the latest release.
+
+For example, a file already sorted by µsort ``1.1.3`` may produce changes when
+sorted by a newer µsort ``1.2.1`` release, but a file already sorted by ``1.2.1``
+should remain unchanged if sorted again by ``1.1.3``. However, a file sorted by
+a future µsort ``2.0.0`` release may not be stable if sorted again by version
+``1.2.1``.
+
+
+Details
+-------
+
+It is our intent to:
+
+1. Bump the post version (and yank the original, depending on how severe) for:
+
+    - documentation or packaging issues, such as missing files in the sdist
+      or unintended classifiers
+
+2. Bump the patch version for:
+
+    - garden-variety bugs, such as those where the previous behavior was an exception
+      (such as `#194 <https://github.com/facebookexperimental/usort/issues/194>`_)
+    - any bug that produces output so mangled it would not have been accepted
+      by human review (such as producing mojibake)
```

Comment:
I think we considered https://github.com/facebookexperimental/usort/pull/187 to be in this category, maybe it's better to give a concrete example.